synapse/config/database.py

```diff
@@ -38,15 +38,17 @@ def read_config(self, config, **kwargs):
 
         self.set_databasepath(config.get("database_path"))
 
-    def generate_config_section(self, data_dir_path, **kwargs):
+    def generate_config_section(self, data_dir_path, database, **kwargs):
+        if not database:
+            database = "sqlite3"
```

Comment:
Given we're passing in a database path, not sure its useful to template out the database type?

Reply:
We're not passing in a database path?

Reply:
Err no, but we are constructing one below

Reply:
I'm missing how we're inferring the database type from the path..

Reply:
If you're using postgres you don't use a path for the database, and you'll most likely have a bunch of other config

Reply:
Can I do something similar to what I've done with listeners: ie, just pass some dict and let the caller deal with it being valid and useful yaml?

Reply:
Yup